### PR TITLE
fix: dont append port to host in case it is already there

### LINKED ISSUE #6: Too many colons
First, this is awesome, thanks for making a Go version of this!

But it doesn't like when I need to specify a port:

```
httpstat "http://45.56.15.67:8680/"
2016/09/23 09:03:11 unable to resolve host: too many colons in address 45.56.15.67:8680:80
```

(Believe me, I wish this was just on port 80 like every other website. But I'm having connectivity issues to it which is why I thought I'd try httpstat, ha.)

### PR DESCRIPTION
**Pull request by ericmdantas**
Closes #6 - not using regexp this time.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -97,7 +97,12 @@ func main() {
 	}()
 
 	t0 := time.Now() // before dns resolution
-	raddr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("%s:%d", host, port))
+
```
Comment (davecheney) [inline main.go]: I'm sorry this fix is not correct. You want to fix the logic inside the function at 87 to split hostport at that point.

There are three cases to deal with

`host`, then return host and choose the port based on the scheme, this is effectively what the code does now

`host:port`, split on the colon, return the first part as the  host, and strconv.Atoi the second part to be the port. There may be a helper function in the net or net/http package that does this.

`:port`: it's an error.
`host:` is also an error

Comment (ericmdantas) [inline main.go]: Thanks!

2beb458 should do the trick.

**Comment by ericmdantas** [comment]
@davecheney I think #13 makes better use of the `stdlib`. Can I close this one?

**Comment by davecheney** [comment]
Yeah, I think @mholt has beaten you to the punchline.

**Pull request closed without merging**